src/AutoRest.CSharp/Mgmt/Generation/ArmResourceExtensionWriter.cs

```diff
@@ -147,44 +135,39 @@ protected override void WriteChildResourceGetMethod(ResourceCollection resourceC
 
         private void WriteResourceEntry(ResourceCollection resourceCollection, bool isAsync, ICollection<FormattableString>? types)
         {
-            if (IsArmCore)
```

Comment:
by design as I described in the description of this PR, Azure.ResourceManager (ArmCore) will never have the `ArmResourceExtension` written, therefore nothing will be output for ArmResourceExtension. We will never go into this method when it is ArmCore, therefore we no longer need this check